### ISSUE TITLE
Postgres; Alter column fails when extending NOT NULL column

### ISSUE THREAD

**Issue by ewesten** [cap-js/cds-dbs#854]
### Description of erroneous behaviour 

When deploying a DB update to postgres we get the following errors.
The corresponding column has been extended in width which should be a compatible change.

Please note that the underlying SQL syntax look invalid for Postgres (we using Postgres 14).

```sh
/> deployment to db:5432 failed
  |
  | error: syntax error at or near "NOT"
  |     at /app/node_modules/pg/lib/client.js:535:17
  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  |     at async Object.run (/app/node_modules/@cap-js/postgres/lib/PostgresService.js:156:26)
  |     at async PostgresService.onPlainSQL (/app/node_modules/@cap-js/postgres/lib/PostgresService.js:293:14)
  |     at async next (/app/node_modules/@sap/cds/lib/srv/srv-dispatch.js:69:17)
  |     at async PostgresService.handle (/app/node_modules/@sap/cds/lib/srv/srv-dispatch.js:67:10)
  |     at async Promise.all (index 1)
  |     at async deploy.schema (/app/node_modules/@sap/cds/lib/dbs/cds-deploy.js:121:3)
  |     at async /app/node_modules/@sap/cds/lib/dbs/cds-deploy.js:39:19
  |     at async Object.to (/app/node_modules/@sap/cds/lib/dbs/cds-deploy.js:38:7) {
  |   length: 92,
  |   severity: 'ERROR',
  |   code: '42601',
  |   detail: undefined,
  |   hint: undefined,
  |   position: '59',
  |   internalPosition: undefined,
  |   internalQuery: undefined,
  |   where: undefined,
  |   schema: undefined,
  |   table: undefined,
  |   column: undefined,
  |   dataType: undefined,
  |   constraint: undefined,
  |   file: 'scan.l',
  |   line: '1176',
  |   routine: 'scanner_yyerror',
  |   query: 'ALTER TABLE BC_VC_ALERT ALTER ENTITY_ID TYPE VARCHAR(400) NOT NULL;\n ^'
  | }
```

### Detailed steps to reproduce

Create CDS model with 

```
entity BC_VC_ALERT {
  key ENTITY_ID : String(100) not null default ' ';
};
```

and use `cds deploy` for deployment  to Postgres. Afterwards, change length from 100 to, say, 400. Run deployment again.

### Details about your project

| Your Project Name |  |
|:------------------|---------------------------------------|
| Node.js version   | v20.x.x                               |
| @sap/cds          | 8.3.1                                 |
| @sap/cds-dk       | 8.3.0                                 |
| @cap-js/postgres | 1.10.1 |

**Comment by simonoswald** [comment]
Hi @ewesten

thank you for reporting this issue! I was able to reproduce it and fix it, I will let you know once we've released a version containing the fix.

Best Regards,
Simon

**Comment by patricebender** [comment]
This has been fixed with `@sap/cds-compiler@5.4.0`:

```
- to.sql.migration: Handle `ALTER COLUMN` for columns with `NOT NULL` and a default value.
```

thanks for reporting :)
